### PR TITLE
Use new environment for object detection components

### PR DESCRIPTION
**Pull request by rdondera-microsoft**
Follow up PR for https://github.com/Azure/azureml-assets/pull/3515 that updates OD/IS components to use the new environment. Tacked on:
1. clean up for input_column_names pipeline input field
2. unit test fixing
3. updates to classification components to reflect an environment fix

Tested by running example notebooks for select tasks:
a. IC, MC: https://ml.azure.com/experiments/id/3cfa7d04-03b8-4d43-a540-b8d6f93e43b4/runs/green_cat_257wd8lr8g?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
b. IC, ML: https://ml.azure.com/experiments/id/948b4383-0829-4aa0-ac57-b6f70e86be0c/runs/tender_yak_z65g7gb498?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
c. IG: https://ml.azure.com/experiments/id/f7a275f7-3dfd-4f69-ae4d-c0701b98d950/runs/loyal_parcel_g7l37yzf1p?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
d. OD: https://ml.azure.com/experiments/id/1a564b75-73f8-4579-a20c-cbe9564cc731/runs/upbeat_kiwi_zm081pmkr0?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
e. IS: https://ml.azure.com/experiments/id/096ae769-8375-4b60-8e38-8cd185bf97a5/runs/cyan_plastic_9s8py4gkg7?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
f. IC, MC, DPV2: https://ml.azure.com/experiments/id/744ce829-14af-4c0a-8a8a-74ad1f5de729/runs/sweet_actor_jz0d55c4cy_HD_0?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47
g. OD, DPV2: https://ml.azure.com/experiments/id/b80967d0-0c6e-4bb7-93a4-2df18bbe424e/runs/serene_town_gqj66htlqt_HD_0?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/validatr&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

Note that the components in the runs above use environments already in production.

### CODE REVIEW

File: assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml [hunk 1]
```diff
@@ -363,7 +363,7 @@ inputs:
 
   input_column_names:
     type: string
-    default: image_url
+    default: image
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml:366]: Why are we changing the default image column name?

Comment (rdondera-microsoft) [reply assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml:366]: To reduce confusion. This change should have been included in https://github.com/Azure/azureml-assets/pull/3016/ , but we missed it. LMK if it makes more sense for this to still be image_url - looks like it according to the existing code, https://github.com/Azure/azureml-assets/blob/main/assets/training/model_evaluation/src/image_dataset.py#L204 .

Comment (rjaincc) [reply assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml:366]: Default in our image FT code is image_url --> 
We might need to update the notebook as well, if we change the col name.

Comment (rdondera-microsoft) [reply assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml:366]: It seems the column name does not actually matter for the FT component. Also, whatever dataset we get from the outside (eg a notebook) is converted, so no need to change the column name. Will chat offline.

Comment (rjaincc) [reply assets/training/finetune_acft_image/components/pipeline_components/hf_classification/spec.yaml:366]: NVM, this is default values. Sorry, I confused it.

**Review by rjaincc — APPROVED** [review]

**Review by rjaincc — DISMISSED** [review]

**Review by gauravrajguru — APPROVED** [review]

**Pull request merged**